### PR TITLE
4201 feat add preset to the calibrate and optimize operator

### PR DESCRIPTION
**Pull request by dvince2**
# Description

* New options exposed for calibrate, as explained by Nelson in issue (this will need some under the hood wiring by @Tom-Szendrey )
* New design to make calibrate and simulate look like optimize
* Standardizing location for a number of enums.
* Relocating the preset option in simulate, request by Pascale.

Note now Simulation and Calibration are done, but Optimize is still TODO

Resolves #(issue)

### CODE REVIEW

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/types/common.ts [hunk 1]
```diff
@@ -22,6 +22,16 @@ export enum DrilldownTabs {
...
 
+export enum CiemssPresetTypes {
+	Fast = 'Fast',
+	Normal = 'Normal'
+}
+
+export enum CiemssMethodOptions {
+	dopri5 = 'dopri5',
+	euler = 'euler'
+}
```
Comment (mwdchang) [inline packages/client/hmi-client/src/types/common.ts:33]: These two are more appropriate under "services/model/simulation-service"

File: packages/client/hmi-client/src/components/workflow/ops/simulate-ciemss/tera-simulate-ciemss.vue [hunk 2]
```diff
@@ -53,7 +55,12 @@
 						</div>
 						<div class="label-and-input">
 							<label for="5">Method</label>
-							<Dropdown id="5" v-model="method" :options="ciemssMethodOptions" @update:model-value="updateState" />
+							<Dropdown
+								id="5"
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/ops/simulate-ciemss/tera-simulate-ciemss.vue:59]: Pretty sure we don't need the "id" part.

File: packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue [hunk 3]
```diff
@@ -349,7 +483,10 @@ const runCalibrate = async () => {
 			mappings: formattedMap
 		},
 		extra: {
-			num_iterations: knobs.value.numIterations
+			num_iterations: state.numIterations,
+			method: state.method,
+			step_size: state.stepSize,
+			learning_rate: state.learningRate
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue:489]: Depend on how the component is setup, which you may need to ask Tom. "state" can be one step behind "knobs" because knobs are the reactive part whereas state came from the parent, so generally speaking we want to send the request with the values embedded under knobs rather than state.

The rough semantic is as such:
- modeller tweak "knobs"
- modeller send request to do stuff
- "knobs" merged back into "state" to sync up parent/child

File: packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/calibrate-operation.ts [hunk 4]
```diff
@@ -60,7 +64,9 @@ export const CalibrationOperationCiemss: Operation = {
 			errorMessage: { name: '', value: '', traceback: '' },
 			numIterations: 100,
 			numSamples: 100,
-			endTime: 100
+			endTime: 100,
+			stepSize: 1,
+			learningRate: 0.1
```
Comment (Tom-Szendrey) [inline packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/calibrate-operation.ts:69]: 0.03 is the default that the pyciemss team uses. We should probably be consistent here as we often ignore the additional settings sections when testing and then get questions about why our results differ

Comment (dvince2) [reply packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/calibrate-operation.ts:69]: thanks!

File: packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue [hunk 5]
```diff
@@ -63,26 +63,98 @@
...
+								v-model="numIterations"
+								@update:model-value="updateState"
+							/>
 						</div>
 						<div class="label-and-input">
 							<label for="num-samples">End time for forecast</label>
 							<InputNumber class="p-inputtext-sm" inputId="integeronly" v-model="knobs.endTime" />
 						</div>
+						<div class="label-and-input">
+							<label for="learning-rate">Step size</label>
```
[the remark this answers is not in the document]
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue:133]: Im not sure what the `for="learning-rate"` in the <label> is used for
Can you fill me in or is it fluff?

**Review by Tom-Szendrey — CHANGES_REQUESTED** [review]
Please give me time to update the java types before merging this.
Corresponding changes will also be made here: https://github.com/DARPA-ASKEM/pyciemss-service/pull/108

**Review by Tom-Szendrey — APPROVED** [review]
I have finished mucking around please test with these pyciemss-service updates:

https://github.com/DARPA-ASKEM/pyciemss-service/pull/109
